flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateCheckpointWriter.java

```diff
@@ -150,6 +151,12 @@ private void complete(boolean precondition, RunnableWithException complete) thro
 	}
 
 	private void finishWriteAndResult() throws IOException {
+		if (inputChannelOffsets.isEmpty() && resultSubpartitionOffsets.isEmpty()) {
+			dataStream.close();
+			result.inputChannelStateHandles.complete(emptyList());
+			result.resultSubpartitionStateHandles.complete(emptyList());
+			return;
+		}
```

Comment:
Don't we have to clean up/remove some empty file? I guess in this branch, `dataStream` will be empty and we have already created some file for it? In other words, won't this code leave a lingering empty files somewhere?

Reply:
No, this is the contract of `CheckpointStateOutputStream`: `closeAndGetHandle` persists the data, while `close` removes the file.